gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/MembershipServiceImpl.java

```diff
@@ -630,38 +630,43 @@ public void deleteReferenceMemberBySource(
         String sourceId
     ) {
         try {
-            final Optional<RoleEntity> optApiPORole = roleService.findByScopeAndName(RoleScope.API, PRIMARY_OWNER.name());
+            RoleEntity apiPORole = roleService
+                .findByScopeAndName(RoleScope.API, PRIMARY_OWNER.name())
+                .orElseThrow(() -> new TechnicalManagementException("Unable to find API Primary Owner role"));
+
             Set<io.gravitee.repository.management.model.Membership> memberships = membershipRepository.findByMemberIdAndMemberTypeAndReferenceTypeAndReferenceId(
                 memberId,
                 convert(memberType),
                 convert(referenceType),
                 referenceId
             );
-            if (!memberships.isEmpty()) {
-                for (io.gravitee.repository.management.model.Membership membership : memberships) {
-                    if (sourceId == null || membership.getSource().equals(sourceId)) {
-                        LOGGER.debug("Delete membership {}", membership.getId());
-                        membershipRepository.delete(membership.getId());
-                        createAuditLog(MEMBERSHIP_DELETED, new Date(), membership, null, environmentId, organizationId);
-                    }
 
-                    if (MembershipReferenceType.APPLICATION.equals(referenceType) && MembershipMemberType.USER.equals(memberType)) {
-                        UserEntity userEntity = findUserFromMembershipMember(
-                            new MembershipMember(memberId, null, memberType),
-                            organizationId,
-                            environmentId
-                        );
-                        applicationAlertService.deleteMemberFromApplication(environmentId, referenceId, userEntity.getEmail());
-                    }
+            for (io.gravitee.repository.management.model.Membership membership : memberships) {
```

Comment:
If there are multiple memberships, and the primary owner is not the first one ;
The endpoint will fail with HTTP 400, but has done half the deletion job. This could be confusing.

Maybe we would better, prior to any deletion process, parse all concerned memberships and check if there is a primary owner.
To ensure this endpoint does nothing if there is one primary owner membership.